### PR TITLE
Animation for flutter/engine#8848

### PR DESCRIPTION
**Pull request by dnfield**
Animation of a mock keyboard showing on a mock iPhone X to demonstrate how padding/viewPadding/insets interact.

### CODE REVIEW

File: packages/diagrams/lib/src/implicit_animations.dart [hunk 1]
```diff
@@ -318,3 +319,97 @@ class AnimatedThemeDiagram extends ImplicitAnimationDiagram<AlignmentGeometry> {
...
 }
+
+class WindowPaddingDiagram extends ImplicitAnimationDiagram<AlignmentGeometry> {
+  const WindowPaddingDiagram({Key key}) : super(key: key);
+
+  @override
+  Curve get curve => Curves.fastOutSlowIn;
+
+  @override
+  Size get size => Size(400.0, 800.0);
```
Comment (gspencergoog) [inline packages/diagrams/lib/src/implicit_animations.dart]:
```suggestion
  Size get size => const Size(400.0, 800.0);
```

File: packages/diagrams/lib/src/implicit_animations.dart [hunk 2]
```diff
@@ -318,3 +319,97 @@ class AnimatedThemeDiagram extends ImplicitAnimationDiagram<AlignmentGeometry> {
...
+
+        // "Keyboard"
+        Align(
+          alignment: AlignmentDirectional.bottomCenter,
+          child: Container(
+            height: selected ? size.height * .4 : 0,
+            width: size.width,
+            child: Container(
+              color: Colors.grey,
+              child: Center(
```
Comment (gspencergoog) [inline packages/diagrams/lib/src/implicit_animations.dart]: Analyzer wants this to be `const`

**Comment by dnfield** [comment]
Cirrus seems to have gone out to lunch on this one :\

**Comment by dnfield** [comment]
@fkorotkov Any ideas on what happened to the CI checks here?  Not sure if it's relevant but I accepted a suggestion through the GitHub web UI and after that Cirrus seemed to not pick this up for testing anymore.

**Comment by fkorotkov** [comment]
@dnfield 👀

**Comment by fkorotkov** [comment]
@dnfield Looks like there was a GitHub API request that failed with an unusual error code which is not retryable. 🤷‍♂ Will investigate further but manually poked the build.

**Comment by dnfield** [comment]
Thanks!  It seems like that `Accept Suggestion` feature could have done that but not sure.

**Pull request merged**